libs/coin-modules/coin-bitcoin/src/__tests__/unit/buildTransaction.fees.unit.test.ts

```diff
@@ -0,0 +1,123 @@
+
+// --------- MOCK ./wallet-btc  ---------
+const estimateAccountMaxSpendable = jest.fn().mockResolvedValue(new BigNumber(123456));
+const buildAccountTx = jest.fn().mockResolvedValue({ fee: 999, inputs: [], outputs: [] });
+
+// swap-able explorer for each test
+let currentExplorer: any = {
+  getNetwork: jest.fn().mockResolvedValue({ relay_fee: "0.00001000" }), // 1 sat/vB
+};
+
+class DummyStrategy {
+  constructor(..._args: any[]) {}
+}
+
+jest.mock("../../wallet-btc", () => ({
+  __esModule: true,
+  default: {
+    estimateAccountMaxSpendable: (...args: any[]) => estimateAccountMaxSpendable(...args),
+    buildAccountTx: (...args: any[]) => buildAccountTx(...args),
+  },
+  getWalletAccount: jest.fn((_account: any) => ({
+    xpub: { explorer: currentExplorer, crypto: {} },
+    derivationMode: "native_segwit",
+  })),
+  CoinSelect: DummyStrategy,
+  DeepFirst: DummyStrategy,
+  Merge: DummyStrategy,
+}));
+
+import { buildTransaction } from "../../buildTransaction";
+
+const BN = (v: BigNumber.Value) => new BigNumber(v);
+const setExplorer = (exp: any) => {
+  currentExplorer = exp;
+};
+
+function makeAccount(currencyId = "bitcoin") {
+  return { id: "acc_1", currency: { id: currencyId } } as any;
+}
+function makeTx(feePerByte: BigNumber.Value, overrides: Partial<any> = {}) {
+  return {
+    recipient: "bc1qexampleaddressxxxxxxxxxxxxxxxxxxxxxx",
+    amount: BN(10000),
+    useAllAmount: false,
+    feePerByte: BN(feePerByte),
+    rbf: true,
+    opReturnData: undefined,
+    utxoStrategy: { strategy: bitcoinPickingStrategy.OPTIMIZE_SIZE, excludeUTXOs: [] },
+    ...overrides,
+  } as any;
+}
+
+beforeEach(() => {
+  jest.clearAllMocks();
+  // default explorer floor = 1 sat/vB
+  setExplorer({ getNetwork: jest.fn().mockResolvedValue({ relay_fee: "0.00001000" }) });
+});
+
+// === UPDATED EXPECTATIONS (NO CLAMPING IN buildTransaction) ===
+
+test("passes user fee 1 sat/vB through unchanged (relay=1)", async () => {
+  const account = makeAccount();
+  const tx = makeTx(1);
+  await buildTransaction(account, tx);
+
+  const fee1 = estimateAccountMaxSpendable.mock.calls[0][1]; // number
+  const fee2 = buildAccountTx.mock.calls[0][0].feePerByte as number; // number
+  expect(fee1).toBe(1);
+  expect(fee2).toBe(1);
+});
+
+test("passes user fee already above floor through unchanged (relay=1, user=3)", async () => {
+  const account = makeAccount();
+  const tx = makeTx(3);
+  await buildTransaction(account, tx);
+
+  expect(estimateAccountMaxSpendable.mock.calls[0][1]).toBe(3);
+  expect(buildAccountTx.mock.calls[0][0].feePerByte).toBe(3);
+});
+
+test("passes fractional fee through unchanged (user=1.2, relay=1)", async () => {
+  const account = makeAccount();
+  const tx = makeTx(1.2);
+  await buildTransaction(account, tx);
+
+  expect(buildAccountTx.mock.calls[0][0].feePerByte).toBe(1.2);
+});
+
+test("ignores higher relay floor when not clamping (relay≈2, user=1 → still 1)", async () => {
```

Comment:
Should we add a test for the clamping if relay=2 and user=1.2 or something like that ?